core/src/main/java/org/apache/lucene/queries/BinaryDocValuesRangeQuery.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.lucene.queries;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.ConstantScoreScorer;
+import org.apache.lucene.search.ConstantScoreWeight;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.Scorer;
+import org.apache.lucene.search.TwoPhaseIterator;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.store.ByteArrayDataInput;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.mapper.RangeFieldMapper;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public final class BinaryDocValuesRangeQuery extends Query {
+
+    private final String fieldName;
+    private final QueryType queryType;
+    private final NormalizedRange expectedRange;
+
+    public BinaryDocValuesRangeQuery(String fieldName, RangeFieldMapper.RangeType rangeType, QueryType queryType, Object from, Object to,
+                                     boolean includeFrom, boolean includeTo) {
+        this.fieldName = fieldName;
+        this.queryType = queryType;
+
+        if (includeFrom) {
+            from = rangeType.nextDown(from);
+        }
+        if (includeTo) {
+            to = rangeType.nextUp(to);
+        }
```

Comment:
This looks a bit backward to me, we would usually rather do
```
if (includeFrom == false) {
    from = rangeType.nextUp(from);
}
```
Do I miss something?